src/components/comparativeD3Graph.js

```diff
@@ -0,0 +1,269 @@
+              .attr('class', 'axisLine')
+              .attr("transform", "translate(30, 0)")
+              .attr('stroke', 'white')
+              .attr('stroke-width', 1)
+              .call(yAxis)
+
+        
+        //create path drawers      
+        const lineGen_current = d3.line()
+              .x(function(d) {
+                  return xScale(d.time);
+              })
+              .y(function(d) {
+                  return yScale(d.todaysTemperature);
+              })
+              .curve(d3.curveCardinal);
+
+        const lineGen_history = d3.line()
+              .x(function(d) {
+                  return xScale(d.time);
+              })
+              .y(function(d) {
+                  return yScale(d.historyTemperature);
+              })
+              .curve(d3.curveCardinal);
+
+        //draw difference                    
+        const line = d3.area()
+              .x(function(d) { return xScale(d.time) })
+              .y(function(d) { return yScale(d.historyTemperature) })
+              .curve(d3.curveCardinal)
+
+        const area = d3.area()
+              .x(function(d) { return xScale(d.time) })
+              .y1(function(d) { return yScale(d.todaysTemperature) })
+              .curve(d3.curveCardinal)
+        
+        vis.datum(dataForRender)
+
+        vis.append('clipPath')
+              .attr('id', 'clip-below')
+              .append('path')
+              .attr('d', area.y0(HEIGHT))
+
+        vis.append('clipPath')
+              .attr('id', 'clip-above')
+              .append('path')
+              .attr('d', area.y0(0))
+
+        vis.append('path')
+              .attr('clip-path', 'url(#clip-above)')
+              .attr('d', area.y0(function(d) { return yScale(d.historyTemperature) }))
+              .attr('transform', 'translate(50, 0)')
+              .attr('fill', 'rgba(78, 183, 213, 0.8)')
+
+        vis.append('path')
+              .attr('clip-path', 'url(#clip-below)')
+              .attr('d', area)
+              .attr('transform', 'translate(50, 0)')
+              .attr('fill', 'rgba(243, 78, 84, 0.85)')
+
+        vis.append('path')
+              .attr('class', 'line')
+              .attr('d', line)
+
+        //draw paths
+        ///current weather
+        vis.append('path')
+              .attr('d', lineGen_current(dataForRender))
+              .attr('class', 'currentWeatherLine')
+              .attr('stroke', '#ef4856')
+              .attr("transform", "translate(50, 0)")
+              .attr('stroke-width', 3)
+              .attr('fill', 'none')
+
+        ///history weather
+        vis.append('path')
+              .attr('d', lineGen_history(dataForRender))
+              .attr('stroke', '#3ed3c7')
+              .attr("transform", "translate(50, 0)")
+              .attr('stroke-width', 3)
+              .attr('fill', 'none')
+
+    }
+  }
+
+  render() {
+    const WIDTH = document.documentElement.clientWidth  
+    const MARGINS = {
```

Comment:
Since you never change MARGINS, it could be made a global constant (e.g. `UPPERCASED_GLOBAL_CONSTANT` in my previous comment)